## Compare GINCO and MT-GINCO labels with FTD

Based on the GINCO and MT-GINCO predictions that we applied to the FTD, I'll analyse how the FTD labels are connected with GINCO labels.

In [2]:
import pandas as pd
import numpy as np

In [4]:
# Import the FTD dataset
ftd = pd.read_csv("data-sheets-with-all-info/FTD-dataset-with-all-information.csv", sep="\t", index_col=0)
ftd.head(3)

ID              labels Multiple labels  \
0  __id__1-syndicate  A1 (argumentative)             NaN   
1  __id__2-syndicate  A1 (argumentative)             NaN   
2  __id__3-syndicate  A1 (argumentative)             NaN   

                                                text  length  \
0  BMW's and Chinese Justice * * * * * In most pl...     975   
1  China and a New Balance of Power SHANGHAI – Th...     956   
2  China and Russia in the New World Disorder Can...     978   

       GINCO_downcast_pred MT-GINCO_downcast_pred  
0           News/Reporting         News/Reporting  
1  Information/Explanation  Opinion/Argumentation  
2    Opinion/Argumentation  Opinion/Argumentation

In [5]:
# Analyse agreement of the labels
ftd["agreement"] = np.where(ftd['GINCO_downcast_pred'] == ftd['MT-GINCO_downcast_pred'], "yes", "no")

ftd["agreement"].value_counts()

yes    1206
no      347
Name: agreement, dtype: int64

In [13]:
347/1206

0.2877280265339967

The predictions of GINCO and MT-GINCO classifier differ in 347 instances (29%).

In [6]:
ftd.columns

Index(['ID', 'labels', 'Multiple labels', 'text', 'length',
       'GINCO_downcast_pred', 'MT-GINCO_downcast_pred', 'agreement'],
      dtype='object')

In [7]:
ftd["Multiple labels"].value_counts()

y    139
Name: Multiple labels, dtype: int64

In [8]:
# Discard texts with multiple labels
ftd = ftd[ftd["Multiple labels"] != "y"]
ftd.describe(include="all")

ID              labels Multiple labels  \
count                1414                1414               0   
unique               1414                  10               0   
top     __id__1-syndicate  A1 (argumentative)             NaN   
freq                    1                 296             NaN   
mean                  NaN                 NaN             NaN   
std                   NaN                 NaN             NaN   
min                   NaN                 NaN             NaN   
25%                   NaN                 NaN             NaN   
50%                   NaN                 NaN             NaN   
75%                   NaN                 NaN             NaN   
max                   NaN                 NaN             NaN   

                                                     text         length  \
count                                                1414    1414.000000   
unique                                               1414            NaN   
top     BMW's and Chinese Justice * * * * * In most pl...            NaN   
freq                                                    1            NaN   
mean                                                  NaN    1445.212164   
std                                                   NaN    4989.570842   
min                                                   NaN      31.000000   
25%                                                   NaN     224.000000   
50%                                                   NaN     495.000000   
75%                                                   NaN    1144.250000   
max                                                   NaN  146922.000000   

            GINCO_downcast_pred   MT-GINCO_downcast_pred agreement  
count                      1414                     1414      1414  
unique                        9                        9         2  
top     Information/Explanation  Information/Explanation       yes  
freq                        407                      311      1118  
mean                        NaN                      NaN       NaN  
std                         NaN                      NaN       NaN  
min                         NaN                      NaN       NaN  
25%                         NaN                      NaN       NaN  
50%                         NaN                      NaN       NaN  
75%                         NaN                      NaN       NaN  
max                         NaN                      NaN       NaN

In [9]:
# Analyse how GINCO primary labels are connected with FTD labels based on the prediction
pd.crosstab(ftd['labels'], ftd["MT-GINCO_downcast_pred"], normalize="index")


MT-GINCO_downcast_pred     Forum  Information/Explanation  Instruction  \
labels                                                                   
A1 (argumentative)      0.016892                 0.162162     0.010135   
A11 (personal)          0.088608                 0.025316     0.012658   
A12 (promotion)         0.015625                 0.109375     0.027344   
A14 (academic)          0.000000                 0.784810     0.012658   
A16 (information)       0.000000                 0.732143     0.017857   
A17 (review)            0.102941                 0.117647     0.014706   
A4 (fiction)            0.000000                 0.074468     0.000000   
A7 (instruction)        0.000000                 0.072727     0.660606   
A8 (news)               0.000000                 0.044118     0.000000   
A9 (legal)              0.000000                 0.205479     0.136986   

MT-GINCO_downcast_pred  Legal/Regulation  List of Summaries/Excerpts  \
labels                                                                 
A1 (argumentative)              0.013514                    0.013514   
A11 (personal)                  0.000000                    0.000000   
A12 (promotion)                 0.000000                    0.023438   
A14 (academic)                  0.012658                    0.012658   
A16 (information)               0.017857                    0.017857   
A17 (review)                    0.000000                    0.029412   
A4 (fiction)                    0.000000                    0.000000   
A7 (instruction)                0.000000                    0.018182   
A8 (news)                       0.022059                    0.080882   
A9 (legal)                      0.506849                    0.000000   

MT-GINCO_downcast_pred  News/Reporting  Opinion/Argumentation     Other  \
labels                                                                    
A1 (argumentative)            0.293919               0.280405  0.111486   
A11 (personal)                0.000000               0.645570  0.227848   
A12 (promotion)               0.066406               0.035156  0.039062   
A14 (academic)                0.037975               0.063291  0.075949   
A16 (information)             0.101190               0.023810  0.029762   
A17 (review)                  0.117647               0.294118  0.000000   
A4 (fiction)                  0.010638               0.106383  0.808511   
A7 (instruction)              0.030303               0.054545  0.030303   
A8 (news)                     0.808824               0.000000  0.029412   
A9 (legal)                    0.013699               0.000000  0.109589   

MT-GINCO_downcast_pred  Promotion  
labels                             
A1 (argumentative)       0.097973  
A11 (personal)           0.000000  
A12 (promotion)          0.683594  
A14 (academic)           0.000000  
A16 (information)        0.059524  
A17 (review)             0.323529  
A4 (fiction)             0.000000  
A7 (instruction)         0.133333  
A8 (news)                0.014706  
A9 (legal)               0.027397

In [12]:
pd.crosstab(ftd['labels'], ftd["MT-GINCO_downcast_pred"], normalize="index").to_dict("index")

{'A1 (argumentative)': {'Forum': 0.016891891891891893,
  'Information/Explanation': 0.16216216216216217,
  'Instruction': 0.010135135135135136,
  'Legal/Regulation': 0.013513513513513514,
  'List of Summaries/Excerpts': 0.013513513513513514,
  'News/Reporting': 0.2939189189189189,
  'Opinion/Argumentation': 0.28040540540540543,
  'Other': 0.11148648648648649,
  'Promotion': 0.09797297297297297},
 'A11 (personal)': {'Forum': 0.08860759493670886,
  'Information/Explanation': 0.02531645569620253,
  'Instruction': 0.012658227848101266,
  'Legal/Regulation': 0.0,
  'List of Summaries/Excerpts': 0.0,
  'News/Reporting': 0.0,
  'Opinion/Argumentation': 0.6455696202531646,
  'Other': 0.22784810126582278,
  'Promotion': 0.0},
 'A12 (promotion)': {'Forum': 0.015625,
  'Information/Explanation': 0.109375,
  'Instruction': 0.02734375,
  'Legal/Regulation': 0.0,
  'List of Summaries/Excerpts': 0.0234375,
  'News/Reporting': 0.06640625,
  'Opinion/Argumentation': 0.03515625,
  'Other': 0.0390625,
  

What we can see based on the GINCO predictions on the FTD labels, is (first, the information for prediction of GINCO-downcast is given, followed by the information what is different on predictions by the MT-GINCO-downcast model):

Most FTD and GINCO-downcast categories match very well, even when we apply the Slovene classifier to the FTD dataset. The only two FTD categories that are not matched well by the GINCO categories are 'A1 (argumentative) and A17 (review). When we apply the MT-GINCO classifier, the results are better for A12 (promotion) (9 points), A4 (fiction) (26 points), A7 (instruction) (6 points), A8 (news) (7 points), but worse for A14 (academic) (3 points), A16 (information) (9 points), A9 (legal) (13 points).

1. Categories that match well:
* 'A11 (personal)': 'Opinion/Argumentation': 0.696
* '': 'Promotion': 0.593; with MT-GINCO better: 'Promotion': 0.683
* 'A14 (academic)': 'Information/Explanation': 0.81; with MT-GINCO slightly worse: 'Information/Explanation': 0.78
* 'A16 (information)': 'Information/Explanation': 0.815; with MT-GINCO slightly worse: 'Information/Explanation': 0.73
* 'A4 (fiction)': 'Other': 0.54; with MT-GINCO much better: 'Other': 0.808
* 'A7 (instruction)': 'Instruction': 0.61, 'Instruction': 0.66
* 'A8 (news)': 'News/Reporting': 0.74, 'News/Reporting': 0.81
* 'A9 (legal)': Legal/Regulation': 0.64, 'Legal/Regulation': 0.51

2. FTD categories that were not identified well:
* 'A1 (argumentative)':  'Information/Explanation': 0.263, 'News/Reporting': 0.246, 'Opinion/Argumentation': 0.229; with MT-GINCO only slightly better: 'Opinion/Argumentation': 0.28,  less Information/Explanation
* 'A17 (review)': 'Promotion': 0.29, 'Information/Explanation': 0.19; with MT-GINCO other categories, but not better: 'Promotion': 0.32, 'Opinion/Argumentation': 0.29

## Compare GINCO labels with CORE

In [3]:
# Import the CORE dataset
core_df = pd.read_csv("data-sheets-with-all-info/CORE-all-information.csv", index_col = 0, sep="\t")

core_df.head(3)

label                                               text  split  \
0  NA OP SR OB  The Top TEN 'Whiniest Sets of Fans' in English...  train   
1        NA NE  Ferry consultation needs deeper questions, say...  train   
2        ID DF  I'v been recording and mixing music for about ...  train   

  main_labels sublabels  Len  main_len  sub_len      GINCORE  \
0       NA OP     SR OB    4         2        2  NA OP SR OB   
1         NaN        NE    2         1        1         News   
2          ID        DF    2         1        1        Forum   

         full_names  main_labels_full_names FTD_pred GINCO_downcast_pred  \
0             SR OB                   NA OP      NaN                 NaN   
1  News Report/Blog               Narrative      NaN                 NaN   
2  Discussion Forum  Interactive Discussion      NaN                 NaN   

  MT-GINCO_downcast_pred  
0                    NaN  
1                    NaN  
2                    NaN

In [4]:
# Filter out only instances that have GINCO prediction
core_df = core_df.dropna(subset = ["GINCO_downcast_pred"])

core_df.shape

(1500, 14)

In [5]:
core_df.columns

Index(['label', 'text', 'split', 'main_labels', 'sublabels', 'Len', 'main_len',
       'sub_len', 'GINCORE', 'full_names', 'main_labels_full_names',
       'FTD_pred', 'GINCO_downcast_pred', 'MT-GINCO_downcast_pred'],
      dtype='object')

In [6]:
core_df

label                                               text  split  \
14     NA SR  Welcome, one and all, to TheCheckingLine.com, ...  train   
16     NA NE  Squeeze - Tommy Squeeze 23 November 2012 When ...  train   
23     NA NE  Independent Travel newsletter: All the latest ...  train   
54     NA NE  Music for More than Just the Ears Photo of emo...  train   
57     NA NE  News Snatch: Discworld, Cube Woes And More Cre...  train   
...      ...                                                ...    ...   
48378  OP OB  Bringing You Great Ideas From The Business Wor...    dev   
48403  NA NE  In the hills outside Islamabad in Pakistan, 16...    dev   
48405  OP OB  Who Cares About the Big Bang? Senator Marco Ru...    dev   
48419  NA NE  McCabe: There was no 'toff strategy' The MP be...    dev   
48423  NA NE  Copeland looking to make statement Miserly for...    dev   

      main_labels sublabels  Len  main_len  sub_len                GINCORE  \
14            NaN        SR    2         1        1                   News   
16            NaN        NE    2         1        1                   News   
23            NaN        NE    2         1        1                   News   
54            NaN        NE    2         1        1                   News   
57            NaN        NE    2         1        1                   News   
...           ...       ...  ...       ...      ...                    ...   
48378          OP        OB    2         1        1  Opinion/Argumentation   
48403         NaN        NE    2         1        1                   News   
48405          OP        OB    2         1        1  Opinion/Argumentation   
48419         NaN        NE    2         1        1                   News   
48423         NaN        NE    2         1        1                   News   

             full_names main_labels_full_names            FTD_pred  \
14        Sports Report              Narrative  A1 (argumentative)   
16     News Report/Blog              Narrative           A8 (news)   
23     News Report/Blog              Narrative  A1 (argumentative)   
54     News Report/Blog              Narrative        A17 (review)   
57     News Report/Blog              Narrative        A17 (review)   
...                 ...                    ...                 ...   
48378      Opinion Blog                Opinion  A1 (argumentative)   
48403  News Report/Blog              Narrative  A1 (argumentative)   
48405      Opinion Blog                Opinion  A1 (argumentative)   
48419  News Report/Blog              Narrative           A8 (news)   
48423  News Report/Blog              Narrative           A8 (news)   

              GINCO_downcast_pred      MT-GINCO_downcast_pred  
14          Opinion/Argumentation       Opinion/Argumentation  
16                 News/Reporting              News/Reporting  
23          Opinion/Argumentation       Opinion/Argumentation  
54                 News/Reporting              News/Reporting  
57     List of Summaries/Excerpts  List of Summaries/Excerpts  
...                           ...                         ...  
48378     Information/Explanation     Information/Explanation  
48403              News/Reporting              News/Reporting  
48405              News/Reporting              News/Reporting  
48419              News/Reporting              News/Reporting  
48423              News/Reporting              News/Reporting  

[1500 rows x 14 columns]

In [7]:
# Discard uninteresting columns
core_df = core_df[['GINCORE', 'sub_len', 'full_names', 'main_len', 'main_labels_full_names','FTD_pred', 'GINCO_downcast_pred', 'MT-GINCO_downcast_pred']]

core_df

GINCORE  sub_len        full_names  main_len  \
14                      News        1     Sports Report         1   
16                      News        1  News Report/Blog         1   
23                      News        1  News Report/Blog         1   
54                      News        1  News Report/Blog         1   
57                      News        1  News Report/Blog         1   
...                      ...      ...               ...       ...   
48378  Opinion/Argumentation        1      Opinion Blog         1   
48403                   News        1  News Report/Blog         1   
48405  Opinion/Argumentation        1      Opinion Blog         1   
48419                   News        1  News Report/Blog         1   
48423                   News        1  News Report/Blog         1   

      main_labels_full_names            FTD_pred         GINCO_downcast_pred  \
14                 Narrative  A1 (argumentative)       Opinion/Argumentation   
16                 Narrative           A8 (news)              News/Reporting   
23                 Narrative  A1 (argumentative)       Opinion/Argumentation   
54                 Narrative        A17 (review)              News/Reporting   
57                 Narrative        A17 (review)  List of Summaries/Excerpts   
...                      ...                 ...                         ...   
48378                Opinion  A1 (argumentative)     Information/Explanation   
48403              Narrative  A1 (argumentative)              News/Reporting   
48405                Opinion  A1 (argumentative)              News/Reporting   
48419              Narrative           A8 (news)              News/Reporting   
48423              Narrative           A8 (news)              News/Reporting   

           MT-GINCO_downcast_pred  
14          Opinion/Argumentation  
16                 News/Reporting  
23          Opinion/Argumentation  
54                 News/Reporting  
57     List of Summaries/Excerpts  
...                           ...  
48378     Information/Explanation  
48403              News/Reporting  
48405              News/Reporting  
48419              News/Reporting  
48423              News/Reporting  

[1500 rows x 8 columns]

In [8]:
# Analyze in how many instance do GINCO and MT-GINCO predictions differ

core_df["agreement"] = np.where(core_df['GINCO_downcast_pred'] == core_df['MT-GINCO_downcast_pred'], "yes", "no")

core_df

/tmp/ipykernel_23915/2950066985.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  core_df["agreement"] = np.where(core_df['GINCO_downcast_pred'] == core_df['MT-GINCO_downcast_pred'], "yes", "no")


GINCORE  sub_len        full_names  main_len  \
14                      News        1     Sports Report         1   
16                      News        1  News Report/Blog         1   
23                      News        1  News Report/Blog         1   
54                      News        1  News Report/Blog         1   
57                      News        1  News Report/Blog         1   
...                      ...      ...               ...       ...   
48378  Opinion/Argumentation        1      Opinion Blog         1   
48403                   News        1  News Report/Blog         1   
48405  Opinion/Argumentation        1      Opinion Blog         1   
48419                   News        1  News Report/Blog         1   
48423                   News        1  News Report/Blog         1   

      main_labels_full_names            FTD_pred         GINCO_downcast_pred  \
14                 Narrative  A1 (argumentative)       Opinion/Argumentation   
16                 Narrative           A8 (news)              News/Reporting   
23                 Narrative  A1 (argumentative)       Opinion/Argumentation   
54                 Narrative        A17 (review)              News/Reporting   
57                 Narrative        A17 (review)  List of Summaries/Excerpts   
...                      ...                 ...                         ...   
48378                Opinion  A1 (argumentative)     Information/Explanation   
48403              Narrative  A1 (argumentative)              News/Reporting   
48405                Opinion  A1 (argumentative)              News/Reporting   
48419              Narrative           A8 (news)              News/Reporting   
48423              Narrative           A8 (news)              News/Reporting   

           MT-GINCO_downcast_pred agreement  
14          Opinion/Argumentation       yes  
16                 News/Reporting       yes  
23          Opinion/Argumentation       yes  
54                 News/Reporting       yes  
57     List of Summaries/Excerpts       yes  
...                           ...       ...  
48378     Information/Explanation       yes  
48403              News/Reporting       yes  
48405              News/Reporting       yes  
48419              News/Reporting       yes  
48423              News/Reporting       yes  

[1500 rows x 9 columns]

In [8]:
core_df.agreement.value_counts()

yes    1235
no      265
Name: agreement, dtype: int64

In [9]:
265/1500

0.17666666666666667

The GINCO and MT-GINCO predictions differ only in case of 265 instances (18% of instances).

### CORE-main

In [11]:
# Analyse how CORE main labels are connected with GINCO labels based on the prediction

# Filter out instances that have multiple main labels
main_labels = core_df[core_df["main_len"] == 1]

pd.crosstab(main_labels["main_labels_full_names"], main_labels["MT-GINCO_downcast_pred"], normalize="index")

MT-GINCO_downcast_pred                    Forum  Information/Explanation  \
main_labels_full_names                                                     
How-To/Instructional                   0.000000                 0.018868   
Informational Description/Explanation  0.007407                 0.381481   
Informational Persuasion               0.022727                 0.068182   
Interactive Discussion                 0.601770                 0.026549   
Lyrical                                0.086957                 0.130435   
Narrative                              0.031949                 0.022364   
Opinion                                0.048148                 0.022222   
Spoken                                 0.000000                 0.043478   

MT-GINCO_downcast_pred                 Instruction  Legal/Regulation  \
main_labels_full_names                                                 
How-To/Instructional                      0.735849          0.000000   
Informational Description/Explanation     0.070370          0.000000   
Informational Persuasion                  0.045455          0.000000   
Interactive Discussion                    0.079646          0.000000   
Lyrical                                   0.000000          0.000000   
Narrative                                 0.000000          0.004792   
Opinion                                   0.059259          0.000000   
Spoken                                    0.000000          0.000000   

MT-GINCO_downcast_pred                 List of Summaries/Excerpts  \
main_labels_full_names                                              
How-To/Instructional                                     0.018868   
Informational Description/Explanation                    0.062963   
Informational Persuasion                                 0.090909   
Interactive Discussion                                   0.017699   
Lyrical                                                  0.000000   
Narrative                                                0.020767   
Opinion                                                  0.025926   
Spoken                                                   0.000000   

MT-GINCO_downcast_pred                 News/Reporting  Opinion/Argumentation  \
main_labels_full_names                                                         
How-To/Instructional                         0.000000               0.056604   
Informational Description/Explanation        0.118519               0.144444   
Informational Persuasion                     0.068182               0.159091   
Interactive Discussion                       0.017699               0.194690   
Lyrical                                      0.000000               0.173913   
Narrative                                    0.616613               0.269968   
Opinion                                      0.096296               0.629630   
Spoken                                       0.304348               0.434783   

MT-GINCO_downcast_pred                    Other  Promotion  
main_labels_full_names                                      
How-To/Instructional                   0.075472   0.094340  
Informational Description/Explanation  0.022222   0.192593  
Informational Persuasion               0.022727   0.522727  
Interactive Discussion                 0.053097   0.008850  
Lyrical                                0.608696   0.000000  
Narrative                              0.017572   0.015974  
Opinion                                0.033333   0.085185  
Spoken                                 0.217391   0.000000

In [12]:
pd.crosstab(main_labels["main_labels_full_names"], main_labels["MT-GINCO_downcast_pred"], normalize="index").to_dict("index")

{'How-To/Instructional': {'Forum': 0.0,
  'Information/Explanation': 0.018867924528301886,
  'Instruction': 0.7358490566037735,
  'Legal/Regulation': 0.0,
  'List of Summaries/Excerpts': 0.018867924528301886,
  'News/Reporting': 0.0,
  'Opinion/Argumentation': 0.05660377358490566,
  'Other': 0.07547169811320754,
  'Promotion': 0.09433962264150944},
 'Informational Description/Explanation': {'Forum': 0.007407407407407408,
  'Information/Explanation': 0.3814814814814815,
  'Instruction': 0.07037037037037037,
  'Legal/Regulation': 0.0,
  'List of Summaries/Excerpts': 0.06296296296296296,
  'News/Reporting': 0.11851851851851852,
  'Opinion/Argumentation': 0.14444444444444443,
  'Other': 0.022222222222222223,
  'Promotion': 0.1925925925925926},
 'Informational Persuasion': {'Forum': 0.022727272727272728,
  'Information/Explanation': 0.06818181818181818,
  'Instruction': 0.045454545454545456,
  'Legal/Regulation': 0.0,
  'List of Summaries/Excerpts': 0.09090909090909091,
  'News/Reporting': 

Comparison of main CORE labels and GINCO-downcast labels

Suprisingly, the main CORE labels are rather well connected to the GINCO-downcast labels, even when the Slovene classifier is used. The only category that is not connected is the category "Spoken".  Some category are better predicted with MT-GINCO ('How-To/Instructional': 'Instruction' - 4 points better; 'Narrative': 'News/Reporting'- 1 point, 'Opinion': 'Opinion/Argumentation' - 8 points, 'Interactive Discussion': 'Forum' - 10 points; 'Lyrical': 'Other' - 13 points), some are worse ('Informational Persuasion': 'Promotion' - 7 points worse; 'Informational Description/Explanation': 'Information/Explanation' - 5 points worse).

1. Well-connected:
* 'How-To/Instructional': 'Instruction': 0.698; with MT-GINCO better: 'Instruction': 0.736
* 'Informational Persuasion': 'Promotion': 0.59; with MT-GINCO worse: 'Promotion': 0.52
* 'Narrative': 'News/Reporting': 0.605; with MT-GINCO slightly better: 'News/Reporting': 0.616
* 'Opinion': 'Opinion/Argumentation': 0.548; with MT-GINCO better: 'Opinion/Argumentation': 0.629

2. A bit less connected:
* 'Informational Description/Explanation': 'Information/Explanation': 0.433; with MT-GINCO worse: 'Information/Explanation': 0.38, 'Promotion': 0.19
* 'Interactive Discussion': 'Forum': 0.495, 'Opinion/Argumentation': 0.22; with MT-GINCO better: 'Forum': 0.60
* 'Lyrical': 'Other': 0.478, 'Opinion/Argumentation': 0.21; with MT-GINCO better: 'Other': 0.61

2. Not well connected (no clear majority label/majority label does not seem to be appropriate):
* 'Spoken': 'News/Reporting': 0.304, 'Opinion/Argumentation': 0.304, 'Other': 0.173; with MT-GINCO similar: 'News/Reporting': 0.30, 'Opinion/Argumentation': 0.43, 'Other': 0.21





### CORE-sub

In [13]:
core_df.head(2)

GINCORE  sub_len        full_names  main_len main_labels_full_names  \
14    News        1     Sports Report         1              Narrative   
16    News        1  News Report/Blog         1              Narrative   

              FTD_pred    GINCO_downcast_pred MT-GINCO_downcast_pred agreement  
14  A1 (argumentative)  Opinion/Argumentation  Opinion/Argumentation       yes  
16           A8 (news)         News/Reporting         News/Reporting       yes

In [17]:
# Analyse how CORE sub labels are connected with GINCO-downcast labels based on the prediction

# Filter out instances that have multiple main labels
sub_labels = core_df[core_df["sub_len"] == 1]

pd.crosstab(sub_labels["full_names"], sub_labels["MT-GINCO_downcast_pred"], normalize="index")

MT-GINCO_downcast_pred              Forum  Information/Explanation  \
full_names                                                           
Advertisement                    0.000000                 0.000000   
Advice                           0.000000                 0.000000   
Course Materials                 0.000000                 0.750000   
Description of a Person          0.000000                 0.448276   
Description of a Thing           0.000000                 0.272727   
Description with Intent to Sell  0.000000                 0.073171   
Discussion Forum                 0.671233                 0.000000   
Editorial                        0.000000                 0.000000   
Encyclopedia Article             0.000000                 0.700000   
FAQ about How-to                 0.000000                 0.000000   
FAQ about Information            0.000000                 0.555556   
Formal Speech                    0.000000                 0.000000   
Historical Article               0.000000                 0.625000   
How-to                           0.000000                 0.020408   
Information Blog                 0.017241                 0.258621   
Interview                        0.000000                 0.055556   
Legal terms                      0.000000                 0.428571   
Letter to Editor                 0.000000                 0.000000   
Magazine Article                 0.000000                 0.000000   
News Report/Blog                 0.020253                 0.002532   
Opinion Blog                     0.064103                 0.019231   
Other Forum                      0.000000                 0.000000   
Other Information                0.000000                 0.200000   
Other Opinion                    0.000000                 0.000000   
Other Spoken                     0.000000                 0.000000   
Personal Blog                    0.076923                 0.019231   
Persuasive Article or Essay      0.200000                 0.000000   
Poem                             0.000000                 0.333333   
Prayer                           0.000000                 1.000000   
Question/Answer Forum            0.475000                 0.100000   
Reader/Viewer Responses          0.500000                 0.000000   
Recipe                           0.000000                 0.166667   
Religious Blogs/Sermons          0.000000                 0.076923   
Research Article                 0.032258                 0.677419   
Reviews                          0.044118                 0.014706   
Short Story                      0.000000                 0.000000   
Song Lyrics                      0.100000                 0.100000   
Sports Report                    0.028302                 0.009434   
TV/Movie Script                  0.000000                 0.000000   
Technical Report                 0.000000                 1.000000   
Technical Support                0.000000                 0.000000   
Transcript of Video/Audio        0.000000                 0.000000   
Travel Blog                      0.090909                 0.000000   

MT-GINCO_downcast_pred           Instruction  Legal/Regulation  \
full_names                                                       
Advertisement                       0.000000          0.000000   
Advice                              0.400000          0.000000   
Course Materials                    0.250000          0.000000   
Description of a Person             0.000000          0.000000   
Description of a Thing              0.045455          0.000000   
Description with Intent to Sell     0.048780          0.000000   
Discussion Forum                    0.041096          0.000000   
Editorial                           0.000000          0.000000   
Encyclopedia Article                0.000000          0.000000   
FAQ about How-to                    0.500000          0.000000   
FAQ about Information               0.222222          0.0

In [18]:
sub_labels["full_names"].value_counts()

News Report/Blog                   395
Opinion Blog                       156
Description of a Thing             132
Sports Report                      106
Personal Blog                      104
Discussion Forum                    73
Reviews                             68
Information Blog                    58
How-to                              49
Description with Intent to Sell     41
Question/Answer Forum               40
Advice                              35
Research Article                    31
Description of a Person             29
Religious Blogs/Sermons             26
Song Lyrics                         20
Encyclopedia Article                20
Interview                           18
Historical Article                  16
Short Story                         11
Travel Blog                         11
FAQ about Information                9
Legal terms                          7
Recipe                               6
Persuasive Article or Essay          5
Other Information        

In [15]:
len(sub_labels.full_names.unique())

43

In [19]:
pd.crosstab(sub_labels["full_names"], sub_labels["MT-GINCO_downcast_pred"], normalize="index").to_dict("index")

{'Advertisement': {'Forum': 0.0,
  'Information/Explanation': 0.0,
  'Instruction': 0.0,
  'Legal/Regulation': 0.0,
  'List of Summaries/Excerpts': 0.0,
  'News/Reporting': 0.0,
  'Opinion/Argumentation': 0.0,
  'Other': 0.0,
  'Promotion': 1.0},
 'Advice': {'Forum': 0.0,
  'Information/Explanation': 0.0,
  'Instruction': 0.4,
  'Legal/Regulation': 0.0,
  'List of Summaries/Excerpts': 0.02857142857142857,
  'News/Reporting': 0.02857142857142857,
  'Opinion/Argumentation': 0.3142857142857143,
  'Other': 0.08571428571428572,
  'Promotion': 0.14285714285714285},
 'Course Materials': {'Forum': 0.0,
  'Information/Explanation': 0.75,
  'Instruction': 0.25,
  'Legal/Regulation': 0.0,
  'List of Summaries/Excerpts': 0.0,
  'News/Reporting': 0.0,
  'Opinion/Argumentation': 0.0,
  'Other': 0.0,
  'Promotion': 0.0},
 'Description of a Person': {'Forum': 0.0,
  'Information/Explanation': 0.4482758620689655,
  'Instruction': 0.0,
  'Legal/Regulation': 0.0,
  'List of Summaries/Excerpts': 0.1034482

If we compare CORE subcategories and GINCO-downcast categories based on the GINCO predictions, we see that 17 CORE subcategories match very well with GINCO categories, 7 match, but less well, and 19 categories do not match well. With some categories, the prediction of MT classifier is better ('Discussion Forum': 'Forum' - 9 points better; 'How-to: 'Instruction' - 4 point better; 'Opinion Blog': 'Opinion/Argumentation' and 'Personal Blog': 'Opinion/Argumentation' - 2 points better; 'Sports Report': 'News/Reporting' - 7 points better; 'Reviews': 'Opinion/Argumentation': 15 points better; 'Song Lyrics': 'Other' - 15 points), in some worse ('Description with Intent to Sell': 'Promotion' - 6 points worse; 'Encyclopedia Article': 'Information/Explanation' - 5 points; 'Historical Article': 'Information/Explanation' - 26 points; 'Persuasive Article or Essay': 'Opinion/Argumentation' - 40 points worse; 'Recipe': 'Instruction' - 16 points worse; 'Travel Blog': 'Opinion/Argumentation' - 19 points worse; 'Legal terms': 'Legal/Regulation': 0.429 on SL, no Legal/Regulation on MT). 

1. Categories that match well:
* 'Advertisement' (1 instance): 'Promotion': 1.0
* 'Course Materials': 'Information/Explanation': 0.75
* 'Description with Intent to Sell': 'Promotion': 0.658; slightly worse with MT: 'Promotion': 0.60
* 'Discussion Forum': 'Forum': 0.575; better with MT: 'Forum': 0.67
* 'Encyclopedia Article': 'Information/Explanation': 0.75; slightly worse with MT: 'Information/Explanation': 0.7
* 'Historical Article': 'Information/Explanation': 0.875; worse with MT:   'Information/Explanation': 0.62, 'Opinion/Argumentation': 0.19
*  'How-to: 'Instruction': 0.63; better with MT: 'Instruction': 0.71
* 'News Report/Blog': 'News/Reporting': 0.79; the same with MT
* 'Opinion Blog': 'Opinion/Argumentation': 0.67; slightly better with MT: 'Opinion/Argumentation': 0.69
* 'Personal Blog': 'Opinion/Argumentation': 0.798; better with MT: 'Opinion/Argumentation': 0.82
* 'Persuasive Article or Essay': 'Opinion/Argumentation': 0.6; much worse with MT: 'Opinion/Argumentation': 0.2
* 'Recipe': 'Instruction': 0.83; worse with MT: 'Instruction': 0.67
* 'Research Article': 'Information/Explanation': 0.677; the same with MT
* 'Sports Report': 'News/Reporting': 0.67; better with MT: 'News/Reporting': 0.74
* 'Technical Report' (1 instance): 'Information/Explanation': 1.0; the same with MT
* 'Technical Support' (1 instance): 'Instruction': 1.0; the same with MT
* 'Travel Blog': 'Opinion/Argumentation': 0.636; worse with MT: 'Opinion/Argumentation': 0.45,'Promotion': 0.27

2. Categories that match, but less well:
* 'Advice': 'Opinion/Argumentation': 0.34, 'Instruction': 0.285; slightly better with MT: 'Instruction': 0.4, 'Opinion/Argumentation': 0.31
* 'Description of a Person': 'Information/Explanation': 0.448, 'Opinion/Argumentation': 0.206; slightly more Opinion with MT
* 'FAQ about How-to': 'Instruction': 0.5, 'Promotion': 0.5; the same with MT
* 'FAQ about Information': 'Information/Explanation': 0.56, 'Instruction': 0.22, 'Promotion': 0.22; the same with MT
* 'Legal terms': 'Legal/Regulation': 0.429, 'Instruction': 0.29; much worse with MT: 'Instruction': 0.43, Information/Explanation': 0.43; 'Legal/Regulation': 0.0
* 'Reviews': 'Opinion/Argumentation': 0.5, 'Promotion': 0.279; much better with MT: 'Opinion/Argumentation': 0.647
* 'Song Lyrics': 'Other': 0.5, 'Opinion/Argumentation': 0.2; better with MT: 'Other': 0.65 

3. CORE sub categories with no (appropriate) majority GINCO-downcast label:
* 'Description of a Thing': 'Information/Explanation': 0.34, 'Promotion': 0.288; more Promotion with MT: 'Promotion': 0.31, 'Information/Explanation': 0.27
* 'Editorial' (2 instances): 'News/Reporting': 1.0; the same with MT
* 'Formal Speech' (2 instances): 'Opinion/Argumentation': 0.5, 'Other': 0.5; MT:  'Opinion/Argumentation': 1.0
* 'Information Blog': 'Information/Explanation': 0.33, 'News/Reporting': 0.19; worse with MT: 'Information/Explanation': 0.26, 'Opinion/Argumentation': 0.21
* 'Interview': 'News/Reporting': 0.44, 'Opinion/Argumentation': 0.22; similar with MT, except more Opinion/Argumentation
* 'Letter to Editor' (1 instance): 'Forum': 1.0; much better with MT: 'Opinion/Argumentation': 1.0
* 'Magazine Article' (3 instances): 'Information/Explanation': 0.33, 'List of Summaries/Excerpts': 0.33, 'News/Reporting': 0.33; different with MT, but not better: 'Opinion/Argumentation': 0.67
* 'Other Forum' (1 instance):  'Opinion/Argumentation': 1.0; the same with MT
* 'Other Information':  'Promotion': 0.4, 'Information/Explanation': 0.2; the same with MT
* 'Other Opinion' (1 instance) 'Promotion': 1.0; the same with MT
* 'Other Spoken' (1 instance): 'Opinion/Argumentation': 1.0; the same with MT
* 'Poem' (3 instances): 'Information/Explanation': 0.33, 'Opinion/Argumentation': 0.33, 'Other': 0.33; same with MT
* 'Prayer' (1 instance): 'Information/Explanation': 1.0; same with MT
* 'Question/Answer Forum': 'Forum': 0.35, 'Other': 0.2; better with MT: 'Forum': 0.475, 'Instruction': 0.15
* 'Reader/Viewer Responses' (2 instances): 'Opinion/Argumentation': 0.5, 'Other': 0.5; better with MT: 'Forum': 0.5, 'Opinion/Argumentation': 0.5
* 'Religious Blogs/Sermons': 'Opinion/Argumentation': 0.30, 'Information/Explanation': 0.27; better (?) with MT: 'Opinion/Argumentation': 0.615
* 'Short Story': 'Opinion/Argumentation': 0.545; much better with MT: 'Other': 0.64
* 'TV/Movie Script' (1 instance): 'Opinion/Argumentation': 1.0; the same with MT
* 'Transcript of Video/Audio' (2 instances): 'Information/Explanation': 0.5, 'Other': 0.5; much better with MT: 'Other': 1.0